### PR TITLE
Splunk KV commands

### PR DESCRIPTION
**Pull request by esharf**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21768

## Description
Adedd the following commands to handle KV stores.
- ***splunk-kv-store-collection-create*** command, which creates a new KV store.
- ***splunk-kv-store-collection-config*** command, which used to configure the KV store fields.
- ***splunk-kv-store-collection-add-entries*** command, which creates a new KV store entry.
- ***splunk-kv-store-collections-list*** command, which print all KV stores.
- ***splunk-kv-store-collection-data-list*** command, which returns the KV store data.
- ***splunk-kv-store-collection-data-delete*** command, which deletes the data in a KV store.
- ***splunk-kv-store-collection-delete*** command, which deleted a KV store.
- ***splunk-kv-store-collection-search-entry*** command, which returns the KV store data for specific query.
- ***splunk-kv-store-collection-delete-entry*** command, which deletes the KV store data with specific query.

## Does it break backward compatibility?
   - [ ] Yes
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by esharf** [comment]
FYI @David-BMS

File: TestPlaybooks/playbook-SplunkPy_KV_commands.yml [hunk 1]
```diff
@@ -0,0 +1,740 @@
+id: SplunkPy_KV_commands
```
Comment (yaakovi) [inline TestPlaybooks/playbook-SplunkPy_KV_commands.yml:1]: move the file to the pack's TestPlaybook folder

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml [hunk 2]
```diff
@@ -376,7 +376,255 @@ script:
...
+      description: The name of the Splunk application that contains the KV store collection.
+        in for example "search"
+      isArray: false
+      name: app_name
+      required: true
+      secret: false
+    deprecated: false
+    description: Add objects to kv store utilizing the batch-save API.
+    execution: false
+    name: splunk-kv-store-collection-add-entries
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:450]: That would be great. In Arcsight we dont have a script so we have to use very complex transformers.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml [hunk 3]
```diff
@@ -376,7 +376,255 @@ script:
...
+  - arguments:
+    - default: false
+      description: The name of the KV store collection
+      isArray: false
+      name: kv_store_collection_name
+      required: true
+      secret: false
+    - default: false
+      description: The list of name and value types to define the KV store collection
+        scheme. For example field.id=number,field.name=string,field.address=string
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:407]: No its not necessary

[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:407]: We also need a better error when the user doesn't provide the field type
Currently its just 
image

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:450]: @David-BMS open an enhancement request for Arcsight

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:407]: @esharf in that case, remove it from the description and fix the error that returns for invalid argument syntax

**Pull request closed without merging**